### PR TITLE
add netty4 metrics.

### PR DESCRIPTION
**Pull request by hujun-w-2**
## What is the purpose of the change
增加netty相关指标，详情：#https://github.com/apache/dubbo/issues/12271

## Brief changelog
**监控指标类型**
- NettyAllocatorMetrics 
- NettyEventExecutorMetrics

**指标详情**
- "netty.allocator.memory.used" - Size of memory used by the allocator, in bytes
Tags: "id" (unique id for the allocator), "allocator.type" (pooled, unpooled), "memory.type" (heap, direct)
- "netty.allocator.memory.pinned" - Size of memory used by allocated buffers, in bytes.
Tags: "id" (unique id for the allocator), "allocator.type" (pooled, unpooled), "memory.type" (heap, direct)
- "netty.allocator.pooled.arenas" - Number of Arenas for a pooled allocator.
Tags: "id" (unique id for the allocator), "allocator.type" (pooled, unpooled), "memory.type" (heap, direct)
- "netty.allocator.pooled.cache.size" - Size of the cache for a pooled allocator, in bytes.
Tags: "id" (unique id for the allocator), "allocator.type" (pooled, unpooled), "cache.type" (normal, small)
- "netty.allocator.pooled.threadlocal.caches" - Number of ThreadLocal caches for a pooled allocator.
Tags: "id" (unique id for the allocator), "allocator.type" (pooled, unpooled)
- "netty.allocator.pooled.chunk.size" - Size of memory chunks for a pooled allocator, in bytes.
Tags: "id" (unique id for the allocator), "allocator.type" (pooled, unpooled)
- "netty.eventexecutor.tasks.pending" - Number of pending tasks in the event executor.
Tags: "name" (unique name for the event executor)

## Verifying this change
image

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by AlbumenJ — COMMENTED** [review]
It is best to add a switch to control whether this ability is enabled

File: dubbo-common/src/main/java/org/apache/dubbo/config/MetricsConfig.java [hunk 1]
```diff
@@ -61,6 +61,11 @@ public class MetricsConfig extends AbstractConfig {
      */
     private Boolean exportMetricsService;
 
+    /**
+     * Enable netty metrics.
+     */
+    private Boolean enableNetty;
```
Comment (songxiaosheng) [inline dubbo-common/src/main/java/org/apache/dubbo/config/MetricsConfig.java:67]: 参考其它配置在dubbo.xsd里面也加入一下，需要支持spring配置

**Review by songxiaosheng — CHANGES_REQUESTED** [review]

**Review by AlbumenJ — CHANGES_REQUESTED** [review]
Please rebase on 3.3 branch, this is a new feature

**Comment by songxiaosheng** [comment]
2023-07-19T14:25:16.4871621Z Error starting ApplicationContext. To display the auto-configuration report re-run your application with 'debug' enabled.
2023-07-19T14:25:16.4872172Z 2023-07-19 14:22:01.284 ERROR 9 --- [           main] o.s.b.d.LoggingFailureAnalysisReporter   : 
2023-07-19T14:25:16.4872403Z 
2023-07-19T14:25:16.4872485Z ***************************
2023-07-19T14:25:16.4872706Z APPLICATION FAILED TO START
2023-07-19T14:25:16.4872918Z ***************************
2023-07-19T14:25:16.4873034Z 
2023-07-19T14:25:16.4873118Z Description:
2023-07-19T14:25:16.4873243Z 
2023-07-19T14:25:16.4873710Z An attempt was made to call the method io.netty.buffer.PooledByteBufAllocator.metric()Lio/netty/buffer/PooledByteBufAllocatorMetric; but it does not exist. Its class, io.netty.buffer.PooledByteBufAllocator, is available from the following locations:
2023-07-19T14:25:16.4874180Z 
2023-07-19T14:25:16.4874530Z     jar:file:/usr/local/dubbo/app/dependency/netty-all-4.0.38.Final.jar!/io/netty/buffer/PooledByteBufAllocator.class
2023-07-19T14:25:16.4874798Z 
2023-07-19T14:25:16.4874919Z It was loaded from the following location:
2023-07-19T14:25:16.4875080Z 
2023-07-19T14:25:16.4875308Z     file:/usr/local/dubbo/app/dependency/netty-all-4.0.38.Final.jar
2023-07-19T14:25:16.4875502Z 
2023-07-19T14:25:16.4875509Z 
2023-07-19T14:25:16.4875583Z Action:
2023-07-19T14:25:16.4875697Z

**Comment by songxiaosheng** [comment]
please fix conflits

**Review by songxiaosheng — APPROVED** [review]
LGTM

**Comment by songxiaosheng** [comment]
@AlbumenJ PTAL

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**